### PR TITLE
Fixes for compiling with PGI

### PR DESCRIPTION
**Pull request by masterleinad**
- In `include/deal.II/base/symmetric_tensor.h` PGI had some problems with the initialization of `Number t[dim][dim];`
- `source/base/utilities.cc` was missing `unistd.h`
- `source/grid/tria_accessor.cc` had a Byte Order Mark pgc++ couldn't deal with

### CODE REVIEW

File: source/grid/tria_accessor.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// ---------------------------------------------------------------------
```
Comment (davydden) [inline source/grid/tria_accessor.cc]: did you add whitespace or you removed it? :smile:

Comment (masterleinad) [inline source/grid/tria_accessor.cc]: I removed that BOM. :smile:

Comment (masterleinad) [inline source/grid/tria_accessor.cc]: This was the only file in `source` and `include` having that.

**Comment by Rombur** [comment]
Nice!

**Comment by drwells** [comment]
That is a surprisingly low number of fixes to get things working with another compiler. 👍

**Comment by masterleinad** [comment]
Well, this is only for a configuration in which all dependencies are disabled...

**Comment by tamiko** [comment]
:+1:

**Pull request merged**